exporters/otlp/src/otlp_http_exporter.cc

```diff
@@ -37,6 +40,208 @@ namespace otlp
+        }
+        cv_.notify_all();
+        break;
+
+      case http_client::SessionState::Created:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: session created" << std::endl;
+        }
+        break;
+
+      case http_client::SessionState::Destroyed:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: session destroyed" << std::endl;
+        }
+        break;
+
+      case http_client::SessionState::Connecting:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: connecting to peer" << std::endl;
+        }
+        break;
+
+      case http_client::SessionState::ConnectFailed:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: connection failed" << std::endl;
+        }
+        cv_.notify_all();
+        break;
+
+      case http_client::SessionState::Connected:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: connected" << std::endl;
+        }
+        break;
+
+      case http_client::SessionState::Sending:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: sending request" << std::endl;
+        }
+        break;
+
+      case http_client::SessionState::SendFailed:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: request send failed" << std::endl;
+        }
+        cv_.notify_all();
+        break;
+
+      case http_client::SessionState::Response:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: response received" << std::endl;
+        }
+        break;
+
+      case http_client::SessionState::SSLHandshakeFailed:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: SSL handshake failed" << std::endl;
+        }
+        cv_.notify_all();
+        break;
+
+      case http_client::SessionState::TimedOut:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: request time out" << std::endl;
+        }
+        cv_.notify_all();
+        break;
+
+      case http_client::SessionState::NetworkError:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: network error" << std::endl;
+        }
+        cv_.notify_all();
+        break;
+
+      case http_client::SessionState::ReadError:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: error reading response" << std::endl;
+        }
+        break;
+
+      case http_client::SessionState::WriteError:
+        if (console_debug_)
+        {
+          std::cerr << "[OTLP HTTP Exporter] Session state: error writing request" << std::endl;
+        }
+        break;
```

Comment:
here too ?

Reply:
In my understanding, some network API will produce a EAGAIN or EWOULDBLOCK error the writing buffer is full, and we could rety to send data a moment later,  it's  also not a failure.